test/utils/updateRateOracle.ts

```diff
@@ -0,0 +1,20 @@
+import { TestTimeAveragedBaseRateOracle, TimeAveragedBaseRateOracle } from 'contracts'
+import { MockProvider } from 'ethereum-waffle'
+import { DAY, timeTravelTo } from '.'
+
+export const updateRateOracle = async (oracle: TimeAveragedBaseRateOracle | TestTimeAveragedBaseRateOracle, cooldown: number, provider: MockProvider) => {
+  const [, timestamps, currIndex] = await oracle.getTotalsBuffer()
+  const newestTimestamp = timestamps[currIndex].toNumber()
+  await timeTravelTo(provider, newestTimestamp + cooldown - 1)
+  await oracle.update()
+}
+
+export const weeklyFillBaseRateOracles = async (tusdOracle: TimeAveragedBaseRateOracle, usdcOracle: TimeAveragedBaseRateOracle, provider: MockProvider) => {
```

Comment:
I think this `weeklyFillBaseRateOracles()` function is too specific to be here. And it is only used in `setupTruefi2.ts`. Consider moving it there or making it more generic.